src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/AbstractApiAction.java

```diff
@@ -201,6 +202,11 @@ protected void handlePut(final RestChannel channel, final RestRequest request, f
 			return;
 		}
 
+		if (isReadonlyFieldUpdated(Utils.convertJsonToJackson(existingConfiguration, false).get(getResourceName()), content)) {
```

Comment:
Delegate conversion from JSON to `isReadonlyFieldUpdated()`, otherwise that conversion impacts all API actions even one that blindly return `false`.